src/puppetlabs/puppetdb/query/report_data.clj

```diff
@@ -0,0 +1,40 @@
+(ns puppetlabs.puppetdb.query.report-data
+  (:require [puppetlabs.puppetdb.jdbc :as jdbc]
+            [puppetlabs.puppetdb.query-eng.engine :as qe]
+            [puppetlabs.puppetdb.scf.storage-utils :as sutils]
+            [puppetlabs.puppetdb.schema :refer [defn-validated]]
+            [schema.core :as s]))
+
+(defn-validated munge-result-rows
+  "Intended to be used for parsing the results
+  from either the report :metrics or :logs queries."
+  [data :- s/Keyword]
+  (fn [rows]
+    (if-let [maybe-json (-> rows first data)]
+      (sutils/parse-db-json maybe-json)
+      [])))
+
+(def valid-query? (comp jdbc/valid-jdbc-query? :results-query))
+(def maybe-sequential? (some-fn nil? sequential?))
```

Comment:
seems like these are really general and belong in a more general ns if we think they're desirable.